tot/plot_utils.py

```diff
@@ -0,0 +1,374 @@
+    Parameters
+    ----------
+    plotting_backend_original : str
+        Original plotting backend.
+
+    Returns
+    -------
+    str
+        The new plotting backend.
+    """
+    if plotting_backend_original == "plotly-auto":
+        plotting_backend_new = "plotly-resample" if validate_current_env() else "plotly"
+    else:
+        plotting_backend_new = plotting_backend_original
+    return plotting_backend_new
+
+
+def validate_plotting_backend_input(plotting_backend):
+    """
+    Validate the input argument for the plotting backend.
+
+    Parameters
+    ----------
+    plotting_backend:str
+        The name of the plotting backend.
+
+    Raises
+    ----------
+    ValueError:
+        If the plotting backend is not a valid backend.
+
+    Returns
+    ----------
+        None
+    """
+    valid_plotting_backends = ["plotly", "plotly-auto", "plotly-resampler"]
+    if plotting_backend in valid_plotting_backends:
+        pass
+    else:
+        log_value_error_invalid_plotting_backend_input()
+
+
+def validate_highlight_forecast_input(highlight_forecast, fcst):
+    """
+    Validate the input argument for the highlight_forecast.
+
+    Parameters
+    ----------
+    highlight_forecast : int
+        The number of forecasts to highlight.
+    fcst : pd.DataFrame
+        The forecast DataFrame.
+
+    Raises
+    ------
+    ValueError
+        If the highlight_forecast value is greater than the number of yhat (prediction horizon) columns in fcst.
+
+    Returns
+    -------
+    None
+    """
+    n_yhat = len([col for col in fcst.columns if "yhat" in col])
+    if highlight_forecast is None:
+        pass
+    elif highlight_forecast <= n_yhat:
+        pass
+    else:
+        log_value_error_invalid_highlight_forecast_input()
+
+
+def validate_df_name_input(df_name, fcst):
+    """
+    Validate the input df_name and returns a dataframe with a single time series and an ID.
+
+    Parameters
+    ----------
+    df_name : str
+        optoinal, ID from time series that should be plotted
+    fcst : pd.DataFrame
+        forecast dataframe
+
+    Returns
+    -------
+    pd.DataFrame
+        A copy of the input dataframe containing the time series data for the specified name.
+
+    Raises
+    ------
+    Exception
+        If the input DataFrame contains more than one time series and the df_name argument is not provided, or if the
+        specified df_name is not present in the DataFrame.
+
+    """
+    fcst, received_ID_col, received_single_time_series, _ = prep_or_copy_df(fcst)
+    if not received_single_time_series:
+        if df_name not in fcst["ID"].unique():
+            assert len(fcst["ID"].unique()) > 1
+            raise Exception(
+                "Many time series are present in the pd.DataFrame (more than one ID). Please, especify ID to be plotted."
+            )
+        else:
+            fcst = fcst[fcst["ID"] == df_name].copy(deep=True)
+            log.info(f"Plotting data from ID {df_name}")
```

Comment:
you can unindent this block as the raise before ensures you would never reach this code unless the condition actually is false
```suggestion
        fcst = fcst[fcst["ID"] == df_name].copy(deep=True)
        log.info(f"Plotting data from ID {df_name}")
```